File: Web-Wicket/src/main/java/com/untitleddoc/cadencecalc/web/wicket/PerimeterClient.java
/*
 * Copyright 2016 Novikov.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.untitleddoc.cadencecalc.web.wicket;

import com.untitleddoc.cadencecalc.jaxrs.models.Perimeter;
import java.util.List;
import javax.ws.rs.ClientErrorException;
import javax.ws.rs.client.Client;
import javax.ws.rs.client.WebTarget;
import javax.ws.rs.core.GenericType;

/**
 * Jersey REST client generated for REST resource:Perimeter [perimeter]<br>
 * USAGE:
 * <pre>
 *        PerimeterClient client = new PerimeterClient();
 *        Object response = client.XXX(...);
 *        // do whatever with response
 *        client.close();
 * </pre>
 *
 * @author Novikov
 */
public class PerimeterClient {

    private final WebTarget webTarget;
    private final Client client;
    private static final String BASE_URI = "http://localhost:8080/jax-rs/webresources";

    public PerimeterClient() {
        client = javax.ws.rs.client.ClientBuilder.newClient();
        webTarget = client.target(BASE_URI).path("perimeter");
    }

    public List<Perimeter> getPerimeter() throws ClientErrorException {
        final var resource = webTarget;
        return resource.request(javax.ws.rs.core.MediaType.APPLICATION_JSON).get(new GenericType<List<Perimeter>>(){});
    }

    public void close() {
        client.close();
    }
    
}

File: Web-Wicket-Kotlin/src/main/kotlin/com/untitleddoc/cadencecalc/web/wicket/kotlin/PerimeterClient.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.untitleddoc.cadencecalc.web.wicket.kotlin

import com.untitleddoc.cadencecalc.jaxrs.models.Perimeter
import javax.ws.rs.client.Client
import javax.ws.rs.client.WebTarget
import javax.ws.rs.core.GenericType;

/*

  @author kisiy
  Created on 2018/02/21
*/

class PerimeterClient {
    private val client: Client
    private val webTarget: WebTarget
    private val BASE_URI = "http://localhost:8080/jax-rs-kotlin/webresources"

    constructor() {
        client = javax.ws.rs.client.ClientBuilder.newClient()
        webTarget = client.target(BASE_URI).path("perimeter")
    }

    public fun getPerimeter() = webTarget.request(javax.ws.rs.core.MediaType.APPLICATION_JSON).get(object : GenericType<List<Perimeter>>() {})
    public fun close() = client.close()
}